src/Illuminate/Database/Eloquent/Collection.php

```diff
@@ -36,6 +36,47 @@ public function find($key, $default = null)
     }
 
     /**
+     * Determine if the given relation is loaded on collection models.
+     *
+     * @param  string  $key
+     * @return bool
+     */
+    public function relationLoaded($key)
+    {
+        // If collection is empty relations are still loaded, but nothing was found
+        if ($this->isEmpty()) {
+            return true;
+        }
```

Comment:
I thought about checking on all models, but was unsure if this would be redundant. I can fix that.